code/fire/_fire.dm

```diff
@@ -1,26 +1,33 @@
 /atom
 	var/on_fire = FALSE
+	var/last_fire_state
+	var/image/fire_overlay
 
-/atom/proc/GetFireIcon()
-	var/image/I
+/atom/proc/GetFireIconState()
+	if(!IsOnFire())
+		return
```

Comment:
Does this return null? I can't remember if `return`ing nothing actually returns null.